sdcm/sct_events.py

```diff
@@ -186,15 +191,33 @@ def __str__(self):
 
 
 class DbEventsFilter(SystemEvent):
-    def __init__(self, type, line=None, node=None):  # pylint: disable=redefined-builtin
+    def __init__(self, type, line=None, node=None, secured_delivery=False):  # pylint: disable=redefined-builtin
         super(DbEventsFilter, self).__init__()
         self.id = id(self)  # pylint: disable=invalid-name
         self.type = type
         self.line = line
         self.node = str(node) if node else None
         self.clear_filter = False
         self.expire_time = None
-        self.publish()
+        if not secured_delivery:
```

Comment:
I don't understand why this needs to be sent multiple times in testing ? We are failing to check for errors somewhere ? And it it happens in testing, it can also happen in real flow, so in theory we should always do that.

Reply:
MQ time to time eat messages, tests are failing due to that.
It most likely due to the worming up proccess in it.
Since tests are failing in 1 of 20 runs, but in real environment it is not happening such often.

Reply:
After going over it again, if we need this functionality it should be part of EventsDevice class, and we should do that every time we start the main device, i.e. not only in testing

Reply:
a function like `EVENTS_PROCESSES['MainDevice'].wait_for_device_ready()` that would replace this
`EVENTS_PROCESSES['MainDevice'].ready_event.wait(timeout=timeout)`, the logic should be similar, i.e. sending some kind of message in the MQ, and waiting for it to pass-throw

Reply:
Great point, fixed accordingly